diff --git a/arch/mips/ath79/mach-db120.c b/arch/mips/ath79/mach-db120.c
index 3fdafb3..6714425 100644
--- a/arch/mips/ath79/mach-db120.c
+++ b/arch/mips/ath79/mach-db120.c
@@ -280,6 +280,11 @@ static void __init db120_setup(void)
 	ath79_eth1_data.speed = SPEED_1000;
 	ath79_eth1_data.duplex = DUPLEX_FULL;
 
+	ath79_gpio_output_select(DB120_GPIO_LED_LAN1,
+                                 AR934X_GPIO_OUT_LED_LINK3);
+  	ath79_gpio_output_select(DB120_GPIO_LED_WAN1,
+                                 AR934X_GPIO_OUT_LED_LINK4);
+
 	ath79_register_eth(1);
 
 	ath79_register_nfc();
